core/src/main/scala/org/apache/spark/Accumulator.scala

```diff
@@ -75,43 +84,65 @@ private[spark] object Accumulators extends Logging {
    * This global map holds the original accumulator objects that are created on the driver.
    * It keeps weak references to these objects so that accumulators can be garbage-collected
    * once the RDDs and user-code that reference them are cleaned up.
+   * TODO: Don't use a global map; these should be tied to a SparkContext at the very least.
```

Comment:
Yep, this is one of the many blockers to multiple active SparkContexts and a source of leaked state in tests, so we should try to have this be scoped within SparkEnv in the future. Feel free to create a followup JIRA for this if you think it's a small task that someone should pick up.